libs/blocks/interlinks/interlinks.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Checks if a given match intersects with an existing match
+ * before adding it to the list of matches. In case of an
+ * intersection, the more specific (i.e. longer) match wins.
+ * @param {array} matches The existing matches
+ * @param {object} contender The match to check and add
+ * @param {number} maxMatches The maximum number of matches
+ */
+export function checkAndAddMatch(matches, contender, maxMatches) {
+  const collisions = matches
+    // check for intersections
+    .filter((match) => !(contender.end < match.start || contender.start > match.end));
+  if (collisions.length === 0 && matches.length < maxMatches) {
+    // no intersecting existing matches, add contender if max not yet reached
+    matches.push(contender);
+  }
+}
+
+/**
+ * Loops through a list of keywords and looks for matches in the article text.
+ * The first occurrence of each keyword will be replaced with a link and tracking added.
+ * The name, location of the keywords file doesn't matter but the file passed in must have a column titled "Keyword".
+ * @param {string} keywordsFilePath The location of the keywords file to be used for interlinks.
+ * @param {number} keywordsLimit The maximum amount of keywords to fetch from the keywords file starting from the top record.  Default is 1000.
+ */
+export default async function interlink(keywordsFilePath, keywordsLimit = 1000) {
```

Comment:
```suggestion
export default async function interlink(path, limit = 1000) {
```
Small suggestion. I like short names and we don't minify.